src/sql/workbench/services/notebook/browser/models/notebookModel.ts

```diff
@@ -290,6 +317,19 @@ export class NotebookModel extends Disposable implements INotebookModel {
 	public async loadContents(isTrusted: boolean = false): Promise<void> {
 		try {
 			this._trustedMode = isTrusted;
+
+			if (!isTrusted) {
+				let whitelistedFolders: string[] = this.workspaceContextService.getValue('notebook.trustedNotebooks');
+				let matchingWhitelistedFolder: string = whitelistedFolders.find(folder => {
+					return this._notebookOptions.notebookUri.fsPath.startsWith(folder);
```

Comment:
So you're doing subfolders too? That definitely isn't obvious to me given the description and is something you need to specifically call out IMO